src/fuse_reduce.cpp

```diff
@@ -210,6 +210,32 @@ static void create_reduce_modules(module_pass_manager& mpm)
     }
 }
 
+namespace {
+MIGRAPHX_PRED_MATCHER(used_once_except_broadcast, instruction_ref ins)
+{
+    if(ins->outputs().size() == 1)
+        return true;
+    if(ins->outputs().size() == 2)
+    {
+        auto non_broadcast =
+            std::find_if(ins->outputs().begin(), ins->outputs().end(), [](instruction_ref output) {
+                return not contains(output->name(), "broadcast");
+            });
+        if(non_broadcast == ins->outputs().end())
+            return false;
+        return std::count_if(
+                   ins->outputs().begin(), ins->outputs().end(), [&](instruction_ref output) {
+                       if(not contains(output->name(), "broadcast"))
+                           return true;
+                       if(output->outputs().size() != 1)
+                           return true;
+                       return output->outputs().front() != *non_broadcast;
```

Comment:
what's the reason for checking output's outputs ? 

name `used_once_except_broadcast` to me suggest that instruction can have more than 2 outputs where all but one are broadcasts.

Reply:
So this is checking that broadcasted output goes into the same instruction as the other output. I rewrote it to make it clearer(the `count_if` is confusing).